### PR TITLE
Die in case of _SKIP_POST_FAIL_HOOKS

### PR DESCRIPTION
**Pull request by asmorodskyi**
In previous implementation of _SKIP_POST_FAIL_HOOKS
die was skipped in case of failure. This patch
improve situation so you can use fatal test flag
together with _SKIP_POST_FAIL_HOOKS

### CODE REVIEW

**Review by SergioAtSUSE — APPROVED** [review]

**Review by Martchus — APPROVED** [review]

**Review by okurz — CHANGES_REQUESTED** [review]
Not sure if it is that easy but could be. Please add corresponding test coverage in t/17-basetest.t

File: t/17-basetest.t [hunk 1]
```diff
@@ -9,6 +9,7 @@ use File::Basename;
 use FindBin '$Bin';
 use Mojo::File 'tempdir';
 use Mojo::Util qw(scope_guard);
+use Test::Exception;
```
Comment (okurz) [inline t/17-basetest.t:12]: please move up to group with the other `Test::…` imports

File: t/17-basetest.t [hunk 2]
```diff
@@ -22,6 +23,29 @@ use needle;
 sub write_with_thumbnail {
 }
 
+subtest run_post_fail_test => sub {
+    my $basetest_class = 'basetest';
+    my $autotest       = Test::MockModule->new("autotest");
+    $autotest->noop('set_current_test');
```
Comment (okurz) [inline t/17-basetest.t:29]: as you just use the mock object once you can combine both lines:

```suggestion
    Test::MockModule->new('autotest')->noop('set_current_test');
```

File: t/17-basetest.t [hunk 3]
```diff
@@ -22,6 +23,29 @@ use needle;
...
+    my $mock_basetest = Test::MockModule->new($basetest_class);
+    $mock_basetest->noop('take_screenshot');
+    $mock_basetest->mock(run => sub { die(); });
+    my $basetest = bless({
+            details      => [],
+            name         => 'foo',
+            category     => 'category1',
+            execute_time => 42,
+    }, $basetest_class);
+
```
Comment (okurz) [inline t/17-basetest.t:39]: I suggest to remove this blank line.

File: t/17-basetest.t [hunk 4]
```diff
@@ -22,6 +23,29 @@ use needle;
...
+    dies_ok { $basetest->runtest } 'run_post_fail end up with die';
+
+    #behavior persists even after define _SKIP_POST_FAIL_HOOKS variable
+    $bmwqemu::vars{_SKIP_POST_FAIL_HOOKS} = 1;
+
+    dies_ok { $basetest->runtest } 'run_post_fail end up with die';
+
```
Comment (okurz) [inline t/17-basetest.t:46]: Remove blank lines in between. Also to make the test strings unique and provide the text you have as in-line comment during runtime I suggest the following:

```suggestion
    $bmwqemu::vars{_SKIP_POST_FAIL_HOOKS} = 1;
    dies_ok { $basetest->runtest } 'run_post_fail behavior persists regardless of _SKIP_POST_FAIL_HOOKS setting';
```

**Comment by asmorodskyi** [comment]
All suggestions applied

**Review by okurz — APPROVED** [review]
Great. Thank you for increasing our test code coverage :)

**Pull request merged**